### PR TITLE
[Enhancement] Add PseudoCluster

### LINKED ISSUE #8212: Fake BE with basic functions to facilitate testing txn publish and clone logic
## Enhancement
Add fake BE with basic functions like tablet, rowset, version, commit, publish etc, so related functions in FE can be tested in UT, like data loading, txn commit/pulish processes, replica clones, etc.

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #8212

## Problem Summary(Required) ：

This PR adds an PseudoCluster, it includes PseudoBackend and PseduFront. PseudoBackend simulates basic BE functions like tablet/txn management, task execution, and reporting, etc. The purpose of this is to facilitate some FE function/feature testing, like load/txn, ddl & dml execution, clone/balance behavior etc.
Currently, it conflict with the existing UT code (mainly because their are too many singletons in FE code), so to use it, the user should invoke a separate new process:

```
mvn compile exec:java -Dexec.mainClass="com.starrocks.pseudocluster.PseudoCluster"

this will create a 3 node pseudo cluster, then connect to this cluster using MySQL client

mysql -h 127.0.0.1 -P 9030 -uroot
// create table & insert into
```

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 5 / 10 (50.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|0|5|00.00%|[120, 121, 129, 130, 131]|
|:large_blue_circle:|com/starrocks/rpc/BrpcProxy.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/ThriftServerEventProcessor.java|1|1|100.00%|[]|

File: fe/fe-core/src/test/java/com/starrocks/pseudocluster/PseudoCluster.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+import com.starrocks.thrift.BackendService;
+import com.starrocks.thrift.HeartbeatService;
+import com.starrocks.thrift.TNetworkAddress;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class PseudoCluster {
```
Comment (nshangyiming) [inline fe/fe-core/src/test/java/com/starrocks/pseudocluster/PseudoCluster.java:21]: So is there any example that show how to use this PseudoCluster in ut code?

Comment (decster) [reply fe/fe-core/src/test/java/com/starrocks/pseudocluster/PseudoCluster.java:21]: Currently, it's not easy to use this in UT, it needs to start a new JVM to reinit some singleton class

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by nshangyiming — APPROVED** [review]

**Pull request merged**